### PR TITLE
refactor: article style

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes FW-2220 FW-2348 FW-2358 FW-2359 FW-2360 FW-2361

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const FARCASTER_REPLY_URL = 'https://relay.farcaster.xyz';
 export const WARPCAST_ROOT_URL = 'https://api.warpcast.com/v2';
 export const WARPCAST_CLIENT_URL = 'https://client.warpcast.com/v2';
 export const FIREFLY_ROOT_URL = env.external.NEXT_PUBLIC_FIREFLY_API_URL;
-export const FIREFLY_DEV_ROOT_URL = 'https://api-dev.firefly.land';
+export const FIREFLY_DEV_ROOT_URL = 'https://api.firefly.land';
```
Comment (guanbinrui) [inline src/constants/index.ts:32]: revert it.

File: src/components/Markup/MarkupLink/NFTCollection.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const { data: collection, isLoading } = useNFTCollection(contractAddress, chainId);
+
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    if (!collection) return null;
+
+    return (
+        <a
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/NFTCollection.tsx:25]: Link

File: src/components/Article/ArticleLayout.tsx [hunk 3]
```diff
@@ -149,7 +150,7 @@ export const ArticleLayout = memo<ArticleLayoutProps>(function ArticleLayout({ a
                         onClick={(event) => event.stopPropagation()}
                         className="block truncate text-clip text-medium leading-5 text-secondary"
                     >
-                        {article.author.handle || ens}
+                        {article.author.handle || ens || formatEthereumAddress(article.author.id, 4)}
```
Comment (guanbinrui) [inline src/components/Article/ArticleLayout.tsx:153]:
```suggestion
                        {article.author.handle || ens || isAddress(article.author.id)}
```

File: src/components/Article/ArticleHeader.tsx [hunk 4]
```diff
@@ -29,36 +27,34 @@ export const ArticleHeader = memo<ArticleHeaderProps>(function ArticleHeader({ a
...
+                    />
+                </Link>
                 <Link
                     href={authorUrl}
                     onClick={(event) => event.stopPropagation()}
-                    className="block truncate text-clip text-medium font-bold leading-5 text-main"
+                    className="block truncate text-clip text-medium leading-5 text-secondary"
                 >
-                    {article.author.handle || ens}
+                    {article.author.handle || ens || formatEthereumAddress(article.author.id, 4)}
```
Comment (guanbinrui) [inline src/components/Article/ArticleHeader.tsx:50]:
```suggestion
                    {article.author.handle || ens || isAddress(article.author.id, 4)}
```

File: src/app/(normal)/article/[id]/pages/DetailPage.tsx [hunk 5]
```diff
@@ -70,6 +75,41 @@ export function ArticleDetailPage({ params: { id: articleId } }: PageProps) {
...
     useDocumentTitle(article ? createPageTitle(t`Post by ${article.author.handle}`) : SITE_NAME);
 
+    useEffect(() => {
+        if (!article?.content) return;
+
+        const images = ref.current?.querySelectorAll('.embed, .zora-embed');
+        if (!images) return;
+
+        images?.forEach((element, index) => {
+            element.addEventListener('click', (event) => {
```
Comment (guanbinrui) [inline src/app/(normal)/article/[id]/pages/DetailPage.tsx:85]: remove those listeners.

File: src/components/Markup/plugins/test.html
Comment (guanbinrui) [inline src/components/Markup/plugins/test.html:1]: Not used

File: src/app/(normal)/article/[id]/pages/DetailPage.tsx (hunk 5 above)
Comment (nuanyang233) [reply src/app/(normal)/article/[id]/pages/DetailPage.tsx:85]: In fact, they are removed due to the event relying on the `index`. Unable to create a function and get its reference to execute removeListener, so I used `replaceChild`.

File: src/components/Article/ArticleHeader.tsx (hunk 4 above)
Comment (nuanyang233) [reply src/components/Article/ArticleHeader.tsx:50]: I need to format, not determine whether it is an address.

File: src/components/Markup/MarkupLink/NFTCollection.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            rel="noopener noreferrer"
+        >
+            <div className="max-w-[250px] rounded-lg bg-bg">
+                <NFTImage
+                    src={collection.banner_image_url}
+                    width={120}
+                    height={120}
+                    className="aspect-square h-full w-full rounded-xl object-cover sm:mb-0 sm:w-auto"
+                    draggable={false}
+                    alt={contractAddress ?? ''}
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/NFTCollection.tsx:39]: Use the NFT collection name as the alt.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**